internal/ent/schema/validator/validate.go

```diff
@@ -34,10 +29,46 @@ func IPBlockPref(prefix string) error {
 	return nil
 }
 
+// PartOfBlock returns error if IP address is NOT part of the block given block's prefix
+func PartOfBlock(ipBlockPref string, ipAdrr string) error {
+	c, _ := cidr.Parse(ipBlockPref)
+	belongsToBlock := false
+
+	c.Each(func(ip string) bool {
+		if ip == ipAdrr {
+			belongsToBlock = true
+		}
+
+		return true
+	})
+
+	if belongsToBlock {
+		return nil
+	}
+
+	return IPAddrOutsideBlockError(ipBlockPref, ipAdrr)
+}
+
+// ErrInvalidIPAddr is an error raised when provided IP Address is invalid
+var ErrInvalidIPAddr = errors.New("provided IP Address is invalid")
+
+// InvalidIPAddrError returns Error Invalid IP Address
+func InvalidIPAddrError(ip string) error {
+	return fmt.Errorf("error %w: %s", ErrInvalidIPAddr, ip)
```

Comment:
not seeing a reason to put `error` in the error text for any of these.

Reply:
also, you should use a [type](https://go.dev/blog/error-handling-and-go#the-error-type) that implements the error interface